pkg/issuer/vault/setup.go

```diff
@@ -0,0 +1,21 @@
+package vault
+
+import (
+	"context"
+
+	"github.com/golang/glog"
+	"github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha1"
+	"k8s.io/api/core/v1"
+)
+
+const (
+	successVaultVerified = "VaultVerified"
+	messageVaultVerified = "Vault verified"
+)
+
+func (v *Vault) Setup(ctx context.Context) error {
+	glog.Info(messageVaultVerified)
+	v.recorder.Event(v.issuer, v1.EventTypeNormal, successVaultVerified, messageVaultVerified)
+	v.issuer.UpdateStatusCondition(v1alpha1.IssuerConditionReady, v1alpha1.ConditionTrue, successVaultVerified, messageVaultVerified)
+	return nil
```

Comment:
We should perform additional checks here if possible - i.e. ensure we can connect to Vault and ensure the `path` is valid for issuing certs.